src/snowflake/connector/cpp/ArrowIterator/CArrowChunkIterator.cpp

```diff
@@ -441,11 +441,13 @@ void DictCArrowChunkIterator::createRowPyObject()
   m_latestReturnedRow.reset(PyDict_New());
   for (int i = 0; i < m_currentSchema->num_fields(); i++)
   {
-    PyObject* value = m_currentBatchConverters[i]->toPyObject(m_rowIndexInBatch);
-    if (value != nullptr)
+    py::UniqueRef value = py::UniqueRef(m_currentBatchConverters[i]->toPyObject(m_rowIndexInBatch));
+    if (!value.empty())
```

Comment:
using `empty` from `UniqueRef` class

Reply:
should the change apply to all usage of toPyObject? E.g. in CArrowChunkIterator::createRowPyObject?

Reply:
Since [PyTuple_SET_ITEM](https://docs.python.org/3/c-api/tuple.html#c.PyTuple_SetItem) steals a reference, it is okay. In the case of _SetItemString it does not steal a reference and hence the need for DECREF'ing / UniqueRefs to properly manage the memory.